feat: implement global lock file management

flferretti opened the pull request:
### Description

<!--- Please include a summary of the change and which issue is fixed. --->
<!--- Please also include relevant motivation and context. -->

This PR adds initial support for a global lockfile for Pixi Global environments. This change introduces a lockfile flow similar to what already exists for workspace projects:
- `install_environment` now returns a computed `LockFile` along with environment updates.
- `sync_environment` also returns a LockFile, ensuring that all global operations expose the lockfile that should be written.
- The CLI (`pixi global …`) is responsible for writing the lockfile to disk as `pixi-global.lock`.

As part of this change, global side effects (exposed executables, shortcuts, completions) are consistently applied after installation or sync, which also resolves a few related issues around global environment behavior.

<!--- Add visual representation of the effect of the change when possible, e.g. before and after screenshots, code snippets, etc. --->

Fixes #1280 
Fixes #3951 

### How Has This Been Tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce while reviewing your changes. -->

Ran the full Python integration test suite for `pixi_global`.
Verified that a `pixi-global.lock` file is produced after global installs/updates, subsequent syncs are no-ops when the lockfile is up-to-date, global environments produce consistent results when re-created using the lockfile.
Performed manual functional checks on installing packages globally, adding/removing exposes and shortcuts, and running `pixi global sync` after editing or removing files.

### AI Disclosure
<!--- Remove this section if your PR does not contain AI-generated content. --->
- [x] This PR contains AI-generated content.
  - [x] I have tested any AI-generated content in my PR.
  - [x] I take responsibility for any AI-generated content in my PR.
<!--- If you used AI to generate code, please specify the tool used and the prompt below. --->
Tools: GitHub Copilot

### Checklist:
<!--- Remove the non relevant items. --->
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added sufficient tests to cover my changes.

flferretti commented:
> Cargo test fails on macOS aarch64. I'm not quite sure if this is caused by my modifications though:
> 
> ```shell
> Error: The template is not valid. .github/workflows/ci.yml (Line: 209, Col: 16): hashFiles('pixi.lock') failed. Fail to hash files under directory '/Users/runner/work/pixi/pixi'
> ```

Fixed after rebasing on main :)

Hofer-Julian reviewed (changes requested):
Thank you for your contribution @flferretti!

I had a quick glance at the code and commented on things that I noticed.

Will now benchmark it locally to see if we'd introduce any performance regressions with this.

flferretti commented:
> @flferretti thanks for the quick follow-up. Seems like CI is red though.
> 
> 
> 
> Personally, I prefer if you'd just append commits instead of force-pushing. We will squash the commits anyway.

Thank you! Sure, if you prefer to squash, I'll just append the commits. 
I'm still working on addressing your comments. I'll drop a message when I'm done. Thank you

flferretti commented:
> I also ran a small benchmark script by Claude and I before you force-pushed.
> 
> These are the results:
> 
> ...
> 
> As a summary, this PR seems to be slightly faster in most cases for the scenarios I checked.
> 
> The full script can be found here: [benchmark.py](https://github.com/user-attachments/files/23855362/benchmark.py)

Thanks a lot @Hofer-Julian, that's very useful! I'll try to re-run the same script once the CI is happy

flferretti commented:
Hi @Hofer-Julian! Just a quick ping to see if you’ve had a chance to look at this :) Let me know if any additional modifications are needed, thank you!
